### PR TITLE
add service account to the preset with workload ID

### PR DESCRIPTION
**Pull request by nawazkh**
add service account to the preset via env var and consume it in the optional azure e2e with workload ID job.

### CODE REVIEW

File: config/jobs/kubernetes-sigs/cluster-api-provider-azure/cluster-api-provider-azure-presets.yaml [hunk 1]
```diff
@@ -14,6 +14,7 @@ presets:
         key: serviceAccountSigningKey
 - labels:
     preset-azure-cred-wi: "true"
+  serviceAccountName: prowjob-default-sa
```
Comment (jsturtevant) [inline config/jobs/kubernetes-sigs/cluster-api-provider-azure/cluster-api-provider-azure-presets.yaml:17]: it looks like we can't use presets for this level of configuration :-(

Comment (jackfrancis) [reply config/jobs/kubernetes-sigs/cluster-api-provider-azure/cluster-api-provider-azure-presets.yaml:17]: so you mean we have to add it directly per-job config?

that's what I see on other jobs fwiw

**Comment by nawazkh** [comment]
running into unmarshalling errors. 
/hold

File: config/jobs/kubernetes-sigs/cluster-api-provider-azure/cluster-api-provider-azure-presets.yaml (hunk 1 above)
Comment (nawazkh) [reply config/jobs/kubernetes-sigs/cluster-api-provider-azure/cluster-api-provider-azure-presets.yaml]: Checking if 
```yaml
spec:
	serviceAccountName: xyz
```
makes any difference.

Comment (nawazkh) [reply config/jobs/kubernetes-sigs/cluster-api-provider-azure/cluster-api-provider-azure-presets.yaml]: Spec will not be parsed. Ref: https://docs.prow.k8s.io/docs/jobs/#presets

Comment (nawazkh) [reply config/jobs/kubernetes-sigs/cluster-api-provider-azure/cluster-api-provider-azure-presets.yaml]: Adding it to per config. Will experiment if adding it to the job config via variable works.

File: config/jobs/kubernetes-sigs/cluster-api-provider-azure/cluster-api-provider-azure-presets.yaml [hunk 2]
```diff
@@ -23,6 +23,8 @@ presets:
     value: "097f89a0-9286-43d2-9a1a-08f1d49b1af8"
   - name: AZURE_FEDERATED_TOKEN_FILE
     value: "/var/run/secrets/azure-token/serviceaccount/token"
+  - name: PROW_SERVICE_ACCOUNT
+    value: "prowjob-default-sa"
```
Comment (nawazkh) [inline config/jobs/kubernetes-sigs/cluster-api-provider-azure/cluster-api-provider-azure-presets.yaml:27]: added `PROW_SERVICE_ACCOUNT` as an env variable to highlight he total set of additions this preset `preset-azure-cred-wi` brings into the test context.

File: config/jobs/kubernetes-sigs/cluster-api-provider-azure/cluster-api-provider-azure-presubmits-main.yaml [hunk 3]
```diff
@@ -117,6 +117,7 @@ presubmits:
     branches:
       - ^main$
     spec:
+      serviceAccountName: $(PROW_SERVICE_ACCOUNT)
```
Comment (jsturtevant) [inline config/jobs/kubernetes-sigs/cluster-api-provider-azure/cluster-api-provider-azure-presubmits-main.yaml:120]: I like this idea, but I am surprise it passed validation

**Comment by jsturtevant** [comment]
/lgtm
We can try this as it only affects this PR job we are validating

**Pull request merged**